docs/content/en/docs/concepts/architecture/components/scheduler/_index.md

```diff
@@ -14,3 +14,152 @@ The Lifecycle Scheduler can trigger events such as
+    metadata:
+      name: podtato-head
+      namespace: podtato-kubectl
+    spec:
+      version: "1.3"
+      revision: 1
+      workloads:
+        - name: podtato-head-left-arm
+          version: 0.1.0
+        - name: podtato-head-left-leg
+          version: 1.2.3
+      preDeploymentTasks:
+        - pre-deployment-hello
+      postDeploymentTasks:
+        - post-deployment-hello
+      preDeploymentEvaluations:
+        - my-prometheus-definition
+      postDeploymentEvaluations:
+        - my-prometheus-definition
+    ```
+
+- **KeptnWorkload**:
+A Workload in Keptn **defines** tasks to be performed during the
+`Pre-Deployment` and `Post-Deployment` phases of a deployment.
+It keeps track of currently active `Workload Instances` responsible
+for performing these checks for specific instances of Deployments,
+StatefulSets, or ReplicaSets.
+
+- **KeptnWorkloadInstance**:
+A Workload Instance **executes** `Pre-Deployment` and `Post-Deployment`
+checks for a specific workload.
+It maintains the status of all checks and the overall state of the `Pre-Deployment` phase.
+Once the desired state is reached (e.g., all pods are up and running), a `PostDeploymentCheck` can be
+triggered.
+
+  For specifying the Pre-Deployment and Post-Deployment checks to be
+  executed for the Workload, use the following annotations:
+
+   `keptn.sh/pre-deployment-tasks: task1,task2`
+
+   `keptn.sh/post-deployment-tasks: task1,task2`
+
+  Similarly, for defining the Evaluations, use the annotations:
+
+   `keptn.sh/pre-deployment-evaluations: my-evaluation-definition`
+
+   `keptn.sh/post-deployment-evaluations: my-eval-definition`
+
+- **KeptnTaskDefinition**:
+The [KeptnTaskDefinition](https://lifecycle.keptn.sh/docs/yaml-crd-ref/taskdefinition/)
+is a CRD used to define tasks that the Keptn Lifecycle Toolkit can run
+during pre- and post-deployment phases.
+Task definitions can be created in the **application's namespace** or the **default** KLT namespace.
+
+- **KeptnTask**:
+A Task is responsible for executing the `TaskDefinition` of a workload.
+It spawns a Kubernetes Job to handle a single Task and keeps track of the Job's status.
+
+- **KeptnEvaluationDefinition**:
+The [KeptnEvaluationDefinition](https://lifecycle.keptn.sh/docs/yaml-crd-ref/evaluationdefinition/)
+is a CRD used to define evaluation tasks that the Keptn Lifecycle Toolkit
+runs during pre- and post-analysis phases.
+Like `KeptnTaskDefinition`, it can be created in the application's namespace or the default KLT namespace.
+
+  A KeptnEvaluationDefinition looks like the following:
+
+  ```yaml
+   apiVersion: lifecycle.keptn.sh/v1alpha3
+    kind: KeptnEvaluationDefinition
+    metadata:
+      name: my-prometheus-evaluation
+    spec:
+      objectives:
+        - keptnMetricRef:
+            name: cpu
+            namespace: ns
+          evaluationTarget: <20
+        - keptnMetricRef:
+            name: memory
+            namespace: ns
+          evaluationTarget: >4
+  ```
+
```

Comment:
Again, this is on the reference page and does not belong in the Architecture section